src/ProxyClient/AbstractProxyClient.php

```diff
@@ -61,58 +50,24 @@
      *                                 requests (optional). This is required if
      *                                 you purge and refresh paths instead of
      *                                 absolute URLs.
-     * @param ClientInterface $client  HTTP client (optional). If no HTTP client
-     *                                 is supplied, a default one will be
-     *                                 created.
-     */
-    public function __construct(array $servers, $baseUrl = null, ClientInterface $client = null)
-    {
-        $this->client = $client ?: new Client();
-        $this->setServers($servers);
-        $this->setBaseUrl($baseUrl);
-    }
-
-    /**
-     * Set caching proxy servers
-     *
-     * @param array $servers Caching proxy proxy server hostnames or IP
-     *                       addresses, including port if not port 80.
-     *                       E.g. array('127.0.0.1:6081')
-     *
-     * @throws InvalidUrlException If server is invalid or contains URL
-     *                             parts other than scheme, host, port
+     * @param HttpClientInterface|ClientInterface $client HTTP client (optional).
+     *                                 If no HTTP client is supplied, a default
+     *                                 one will be created.
      */
-    public function setServers(array $servers)
+    public function __construct(array $servers, $baseUrl = null, $client = null)
     {
-        $this->servers = array();
-        foreach ($servers as $server) {
-            $this->servers[] = $this->filterUrl($server, array('scheme', 'host', 'port'));
+        if ($client instanceof ClientInterface) {
+            // Only for BC; Guzzle 3 for PHP 5.3 compatibility
+            $this->client  = new Guzzle3Adapter($client);
+        } elseif ($client instanceof HttpClientInterface) {
+            $this->client = $client;
+        } elseif (null === $client) {
+            $this->client = new Guzzle3Adapter();
```

Comment:
You mean defaulting to Guzzle 5 (or Buzz) when we find that? Makes sense to much this logic into a small factory, then.